### PR TITLE
LPAL-253 details element keyboard navigation fixes

### PR DESCRIPTION
**Pull request by townxelliot**
## Purpose

Ensure that links and inputs inside `<details>` elements can be accessed via keyboard navigation when the element is expanded.

Fixes LPAL-253

## Approach

Rewrite the polyfill to correct its behaviour

Add a cypress test to expand and navigate the content of a `<details>` element (and the same for a polyfilled element behaving as a details element).

## Learning

Possibly should have used the govuk-frontend rewrite of the polyfill, rather than do it myself. This rewrite does have the advantage that you can apply it to elements other than `<details>`, which makes it easier to functional test (see enclosed cypress test). Also, I don't think the govuk-frontend polyfill attaches ARIA attributes (which this implementation does).

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [x] The product team have tested these changes

### CODE REVIEW

**Review by nickdavis2001 — DISMISSED** [review]
lgtm

**Comment by townxelliot** [comment]
Apologies @nickdavis2001, after merging master it dismissed your review. Could you please approve again?

**Review by nickdavis2001 — APPROVED** [review]

**Review by williamfalconeruk — APPROVED** [review]

**Review by nickdavis2001 — APPROVED** [review]

**Pull request merged**
